app/shop.py

```diff
@@ -0,0 +1,27 @@
+import datetime
+
+
+class Shop:
+    def __init__(self, name: str, location: list, products: dict) -> None:
```

Comment:
The parameter 'location' is typed as a list, but it might be more descriptive to specify the type of elements in the list, for example, List[str] or List[float], depending on what 'location' is supposed to represent.